### PR TITLE
Respect .dockerignore files

### LINKED ISSUE #224: [BUG] dockerignore not respected
**Describe the bug**
We don't currently take `.dockerignore` files into account when creating the build context.

### PR DESCRIPTION
**Pull request by brandonbloom**
Fixes #224

### CODE REVIEW

File: internal/providers/docker/components/container/build.go [hunk 1]
```diff
@@ -238,3 +257,47 @@ func tarBuildContext(w io.WriteCloser, root string) (err error) {
 	}
 	return nil
 }
+
+func createIgnoreMatcher(buildContext string) (*fileutils.PatternMatcher, error) {
+	// Initialize patterns with always ignored files.
+	const ignoreBasename = ".dockerignore"
+	patterns := []string{
+		ignoreBasename,
```
Comment (BenElgar) [inline internal/providers/docker/components/container/build.go:265]: Why are we including `.dockerignore` in the list of ignored files? Docker doesn't:

```
❯ cat Dockerfile
FROM bash
COPY . .

❯ cat .dockerignore
my-file

❯ docker run $(docker build -q .) cat .dockerignore
my-file
```

Or am I misunderstanding what's happening?

File: internal/providers/docker/components/container/build.go [hunk 2]
```diff
@@ -238,3 +257,47 @@ func tarBuildContext(w io.WriteCloser, root string) (err error) {
 	}
 	return nil
 }
+
+func createIgnoreMatcher(buildContext string) (*fileutils.PatternMatcher, error) {
+	// Initialize patterns with always ignored files.
+	const ignoreBasename = ".dockerignore"
```
Comment (BenElgar) [inline internal/providers/docker/components/container/build.go:263]: I'll note, just as an FYI, that docker also supports ignore files of the format `<dockerfile-name>.dockerignore`. Though you have to enable Buildkit for that to work. It's reasonably widely used and I've used the feature in every non-trivial docker project I've worked on, including the console.

https://stackoverflow.com/a/57774684/1142314
https://github.com/deref/deref/blob/master/registry/build.Dockerfile.dockerignore

**Review by BenElgar — APPROVED** [review]

File: internal/providers/docker/components/container/build.go (hunk 1 above)
Comment (brandonbloom) [reply internal/providers/docker/components/container/build.go]: I was copying what I saw in moby/moby, but I forget where. Maybe I misread it.

File: internal/providers/docker/components/container/build.go (hunk 2 above)
Comment (brandonbloom) [reply internal/providers/docker/components/container/build.go]: Do we have a ticket somewhere to support Buildkit builds?

**Pull request merged**